e2e/_suites/fleet/features/fleet_server.feature

```diff
@@ -0,0 +1,19 @@
+@fleet_server
+Feature: Fleet Server
+  Scenarios for Fleet Server, where an Elasticseach and a Kibana instances are already provisioned,
+  so that the Agent is able to communicate with them
+
+@start-fleet-server
+Scenario Outline: Deploying the <os> fleet-server agent
+  When a "<os>" agent is deployed to Fleet with "tar" installer in fleet-server mode
+  Then Fleet server is enabled
```

Comment:
@EricDavisX  @blakerouse I wrote this `Then` clause, but I'd like to know if there is an assertion that should be done here. Something like: `the elastic-agent process is started`, or `the Fleet app in Kibana shows "FooBar" in the Fleet page`, or "elasticsearch contains THIS doc in THAT index".

Could you help me here in writing the right expected behaviour?

Reply:
I am not too familiar with the structure of these files so I might not be fully understanding the context.

I believe your asking if its possible to known if the Fleet Server is running correct. The simplest way is to check that the Agent is reported Healthy in Kibana. That might seem to simple but the **only** way for the Agent running a Fleet Server to show in Kibana as healthy is if it can communicate to its local Fleet Server and that Fleet Server can write to elasticsearch.

Reply:
Never mind about the code structure yet, I'm still interested in the behavior of the product without considering internal details/implementations.

With that in mind:

- what ES query should we write to verify that?
- is it enough to check the healthy status for the promoted-to-server agent?

Reply:
The check can be the same as for all Agents, that it is listed as 'healthy' in the Agents API list call.  A secondary check could be to assess if the Fleet Server process is running on the host, as noted in this example:

```
edavis-mbp:elastic-agent-8.0.0-SNAPSHOT-darwin-x86_64-infra-build edavis$ ps ax | grep elastic
SNAPSHOT-darwin-x86_64/fleet-server --agent-mode -E logging.level=info -E http.enabled=true -E http.host=unix:///Library/Elastic/Agent/data/tmp/default/fleet-server/fleet-server.sock -E logging.json=true -E logging.ecs=true -E logging.files.path=/Library/Elastic/Agent/data/elastic-agent-53d75c/logs/default -E logging.files.name=fleet-server-json.log -E logging.files.keepfiles=7 -E logging.files.permission=0640 -E logging.files.interval=1h -E path.data=/Library/Elastic/Agent/data/elastic-agent-53d75c/run/default/fleet-server--8.0.0-SNAPSHOT
```
This example is from macOS, but the the process name is the same.